design/parallel-commit/globally-non-unique-timestamps.md

```diff
@@ -1,3 +1,27 @@
+# Allow commit_ts to be non-globally unique.
+
+This document is about the problem and the solution of the non-globally-unique `commit_ts` problem.
+
+## The problem
+
+in TiKV, rolling back a transaction need to leave a rollback record in order to guarantee consistency. But rollback records and commit records are both saved in Write CF, and the commit records are saved with `{user_key}{commit_ts}` as the internal key, while that of rollback records are `{user_key}{start_ts}`. Previously the `commit_ts`es are TSOs allocated from PD, which is guaranteed globally unique. However when we tries to use a calculated timestamp as the `commit_ts` to avoid the latency of PD's RPC, it's possible that a rollback record gets the same internal key as a commit record, but we need to keep them both.
```

Comment:
```suggestion
In TiKV, rolling back a transaction need to leave a rollback record in order to guarantee consistency. But rollback records and commit records are both saved in Write CF, and the commit records are saved with `{user_key}{commit_ts}` as the internal key, while that of rollback records are `{user_key}{start_ts}`. Previously the `commit_ts`es are timestamps allocated from PD, which is guaranteed globally unique. However when we try to use a calculated timestamp as the `commit_ts` to avoid the latency of PD's RPC, it's possible that a rollback record gets the same internal key as a commit record, but we need to keep them both.
```
TSO refers to the timestamp allocator